scripts/main.py

```diff
@@ -1,11 +1,21 @@
+import argparse
+
+import nisapi
 import yaml
+
 import iup
-import argparse
 
 
-def run(config):
+def run(config: dict, cache: str):
+    data = nisapi.get_nis(cache)
+
     # List of the cumulative data sets described in the yaml
-    cumulative_data = [iup.parse_nis(**x) for x in config["data"].values()]
+    # note that df.filter(**{"column1": value1, "column2": value2}) is equivalent to
+    # df.filter(pl.col("column1")==value1, pl.col("column2")==value2)
+    cumulative_data = [
+        iup.CumulativeUptakeData(data.filter(**x["filters"]).collect())
+        for x in config["data"].values()
+    ]
 
     # List of grouping factors used in each data set
     grouping_factors = iup.extract_group_names(
```

Comment:
`extract_group_names` had accepted a list of dictionaries - the how-to-rename-grouping-columns dictionaries from each data set. But this is no longer necessary. `extract_group_names` should now accept a list of lists - the names-of-grouping-columns from each data set.

Reply:
I think now we will have the same grouping columns from every dataset, since the data schema is the same. It should always be `geography` and `domain`, right? (This should still be a function argument, but I think we just need a list, because we can assume that all incoming datasets have the same schema.)

The `"filters"` bit I think you still want, because a "dataset" is defined by taking the whole pile of NIS data and subsetting based on vaccine, indicator, etc.

Although I'm now thinking we might want more sophisticated filters, like start and end dates? Or there could be some preprocessor that takes the NIS data, pulls out just the uptake indicator, labels by season, etc. Because I think that's what we're going to want for all the models

Reply:
Yes, I think you are right that "geography" and "domain" are the only grouping factor column names possible. So all `extract_group_names` must do is determine which/neither/both of these should be used consistently across all data sets being used together. See Issue #78.

I'm not sure we need more sophisticated pre-processing, because the "filters" entry in the config will pull out just the uptake indicator, and `date_to_season` can already label by season.

Reply:
I realized that you only need to filter `indicator` for `"received a vaccination"`, which is consistent across datasets. So we can drop `indicator_type` and `indicator`. That's what's in `preprocess.py` for now; I expect we'll refactor a lot of this later.